index.js

```diff
@@ -370,20 +370,67 @@ function isWebcam(device) {
     device.deviceDescriptor.bDeviceProtocol === 0x01
 }
 
+function getInterfaces(device) {
+  const interfaces = Array.isArray(device.interfaces) ? device.interfaces : [];
```

Comment:
We should create a copy of the interfaces array to not change the original object.
```
const interfaces = Array.isArray(device.interfaces) ? [...device.interfaces] : [];
```